packages/chats/src/Lightbox/hooks/useDimensions.native.tsx

```diff
@@ -0,0 +1,52 @@
+import type { View } from 'react-native';
+import Animated, { useSharedValue, useAnimatedReaction, measure } from 'react-native-reanimated';
+import type { Dimensions } from '../types';
+import { DuplicateContentState } from '../constants';
+
+export const useDimensions = (
+    originalRef: React.RefObject<View>,
+    duplicateContentState: Animated.SharedValue<DuplicateContentState>,
+    onMeasureEnd: () => void,
+): Dimensions => {
+    const width = useSharedValue<number>(0);
+    const height = useSharedValue<number>(0);
+    const pageX = useSharedValue<number>(0);
+    const pageY = useSharedValue<number>(0);
+
+    useAnimatedReaction(
+        () => {
+            return {
+                duplicateContentState: duplicateContentState.value,
+                originalRef,
+            };
+        },
+        (state, prevState) => {
+            try {
+                if (
+                    (!prevState ||
+                        prevState.duplicateContentState !== DuplicateContentState.Measurement) &&
```

Comment:
Let's use `@inline` variables instead `enum`s to decrease memory consumption. Please check all such places where they are inside `worklets`, I won't mark each one :)